setendings.py

```diff
@@ -0,0 +1,397 @@
+# on the nodes.
+# TODO: Separate into Class? .. would need static methods.
+################################################################################
+
+def _set_end_lineno(node):
+    """Transform the property in-place.
+    Testing with simple, single-line things for now.
+    Goal is to work on various multi-line nodes, with complicated nesting.
+    'tolineno' is set correctly on everything, by asteroid.
+    """
+    # Guard. 
+    if node.tolineno is None:
+        # TODO: write new code to gracefully handle, if this is ever reached.
+        raise Exception('''ERROR:️ lineno is None at node:''', node, 
+                        node.as_string())
+    else:
+        node.end_lineno = node.tolineno
+
+def _set_end_col_offset_by_string(node):
+    """Nodes without children can get end_col_offset by length of string.
+    Hopefully this works without problems.
+    """
+    NO_CHILDREN_TYPE = [astroid.Arguments, astroid.AssignName, astroid.Break, 
+            astroid.Const, astroid.DelName, astroid.Ellipsis, astroid.Global, 
+            astroid.Import, astroid.ImportFrom, astroid.List, astroid.Name, 
+            astroid.Nonlocal, astroid.Pass, astroid.Yield]
+
+    node_string = node.as_string()
+
+    if type(node) not in NO_CHILDREN_TYPE:
+        raise Exception('''ERROR:️ node not found in NO_CHILDREN list.''', node, 
+                        node_string)
+
+    # Guard: to use this approach, it shouldn't have children.
+    for child in node.get_children():  # get_children() returns a generator.
+        raise Exception('''ERROR:️ should not be called if has children.''', 
+                        node, node_string)
+
+    # Guard: should have .value property
+    if not hasattr(node, 'as_string'):
+        raise Exception('''ERROR:️ node must have the .as_string property.''', 
+                        node, node_string)
+
+    # _set_child_end_col_offset_by_value(node)
+    node.end_col_offset = node.col_offset + len(node_string)
+
+def _set_end_col_offset(node):
+    """Set the end_col_offset property by the as_string attribute.
+    Uses the end column location of the last line, instead of the widest line.
+    Need to do a custom postorder traversal to set end_col_offset property of 
+    child nodes first, since it is sometimes unset in the last child.
+    Postorder setting of children results in the assumption: if a node has the
+    propert set, then all of its children are set.
+    """
+
+    # Compare with "None" since value could be "0", which is falsey.
+    # Returning a.s.a.p. reduces the runtime.
+    # Some nodes dont have the attribute set yet.
+    if hasattr(node, 'end_lineno') and hasattr(node, 'end_col_offset'):
+        if node.end_lineno is not None and node.end_col_offset is not None:
+            return  # since properties are already set.
+
+    # Doing this first makes it a postorder traversal.
+    for child in node.get_children():  # get_children() returns a generator.
```

Comment:
Please remove this.

The postorder traversal happens "automatically" when the TransformVisitor does its thing.

Of course, it will only call the function on the nodes for which it has been registered. I hope I made clear today that the only nodes in the AST that I care about right now are `Const` and `BinOp`, which are the ones in your tests anyway (so they should still pass).